cunumeric/array.py

```diff
@@ -184,6 +186,140 @@ def _convert_all_to_numpy(obj: Any) -> Any:
         return obj
 
 
+def writeable() -> Callable[[Callable[P, R]], Callable[P, R]]:
```

Comment:
should `writable()` take a list of arrays as an argument? I imagine the case when some of the arrays passed to the method can be writable and some not.

Reply:
From what I saw on the ndarray APIs,  the write access on numpy APIs should be checked on two cases. 
1. The array where the corresponding method is calling to change the array in place. 
2. the `out` array passed to `out` argument. 

Case 1 is checked by `writeable` and case 2. is changed by `add_boilerplate` when the decorator checks the kwargs.

Reply:
I'm ambivalent about a decorator for this. It only appears to be used in ~4 places, do we expect it to apply to many more places in the future? It seems another alternative could be a helper predicate `check_writeable(...)` that could be called inside the few methods that need to check whether inputs are writeable. A decorator would seem more justified to me if it e.g. automatically added information to the docstring to document the writeable constraint.  If we do keep the decorator, I think `check_writeable` might be a better name, since it is more descriptive of what the decorator does, and also distinguishes the name from the "writeable" property on `FlagsObj` a bit more directly.

Reply:
Any newly implmented cuNumeric API which does writing to the underlying array will need this decorator. As you mentioned, we already have many decorators, I tried to apply this decorator to only corresponding cases. 
I agree with you on your suggested decorator name.

Reply:
Can you see what it looks like with a `check_writeable` function called inside the methods? My concern is that after 2- levels of decoration, it becomes difficult to reason effectively about what a function does, in isolation. Accordingly, I think it would be better to reserve the use of new decorators for things that can *only* be accomplished with decorators (which does not seem to be the case here)

Reply:
OK. I'll change the code to use a function instead of a decorator as you suggested.